File: library/src/main/java/name/zeno/android/third/glide/sizableurl/SizableUrlModel.java
package name.zeno.android.third.glide.sizableurl;

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2017/3/10.
 */
public abstract class SizableUrlModel
{
  protected String baseUrl;

  abstract String requestCustomSizeUrl(int width, int height);

  @Override public boolean equals(Object obj)
  {
    if (obj instanceof ZQiniuUrl) {
      ZQiniuUrl other = (ZQiniuUrl) obj;
      return (baseUrl + "").equals(other.baseUrl);
    }
    return false;
  }

  @Override public int hashCode()
  {
    return (baseUrl + "").hashCode();
  }
}

File: library/src/main/java/name/zeno/android/third/glide/sizableurl/SizableUrlModel.kt
package name.zeno.android.third.glide.sizableurl

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2017/3/10.
 */
abstract class SizableUrlModel {
  protected var baseUrl: String? = null

  internal abstract fun requestCustomSizeUrl(width: Int, height: Int): String?

  override fun equals(other: Any?): Boolean {
    if (other is ZQiniuUrl) {
      return baseUrl == other.baseUrl
    }
    return false
  }

  override fun hashCode(): Int {
    return "$baseUrl".hashCode()
  }
}
